modules/detection_cluster.py

```diff
@@ -0,0 +1,39 @@
+"""
+LiDAR detection cluster data structure.
+"""
+
+from . import lidar_detection
+
+
+class DetectionCluster:
+    """
+    Cluster of LiDAR detections.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(
+        cls, detections: "list[lidar_detection.LidarDetection]"
```

Comment:
since it's not possible to return None here, you should change the type annotation to reflect that
sidenote just remember that our design pattern has null checking before passing to create so detections should always exist